cmd/pg/daemon.go

```diff
@@ -0,0 +1,55 @@
+package pg
+
+import (
+	"os"
+	"path"
+
+	"github.com/spf13/cobra"
+	"github.com/wal-g/tracelog"
+	"github.com/wal-g/wal-g/internal"
+	"github.com/wal-g/wal-g/internal/asm"
+	"github.com/wal-g/wal-g/internal/databases/postgres"
+	"github.com/wal-g/wal-g/utility"
+)
+
+const DaemonShortDescription = "Uploads a WAL file to storage"
+
+// daemonCmd represents the daemon archive command
+var daemonCmd = &cobra.Command{
+	Use:   "daemon daemon_socket_path",
+	Short: DaemonShortDescription, // TODO : improve description
+	Args:  cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		uploader, err := postgres.ConfigureWalUploader()
+		tracelog.ErrorLogger.FatalOnError(err)
+
+		archiveStatusManager, err := internal.ConfigureArchiveStatusManager()
+		if err == nil {
+			uploader.ArchiveStatusManager = asm.NewDataFolderASM(archiveStatusManager)
+		} else {
+			tracelog.ErrorLogger.PrintError(err)
+			uploader.ArchiveStatusManager = asm.NewNopASM()
+		}
+
+		PGArchiveStatusManager, err := internal.ConfigurePGArchiveStatusManager()
+		if err == nil {
+			uploader.PGArchiveStatusManager = asm.NewDataFolderASM(PGArchiveStatusManager)
+		} else {
+			tracelog.ErrorLogger.PrintError(err)
+			uploader.PGArchiveStatusManager = asm.NewNopASM()
+		}
+
+		uploader.UploadingFolder = uploader.UploadingFolder.GetSubFolder(utility.WalPath)
+		pathToData, exist := os.LookupEnv("PGDATA")
```

Comment:
We use viper for configuration:

https://github.com/wal-g/wal-g/blob/71fb7e9d110a80d971744963ac5c2cabfd041cc3/tests_func/helpers/walg.go#L124